lib/web_ui/lib/src/engine/view_embedder/embedding_strategy/embedding_strategy.dart

```diff
@@ -21,13 +18,7 @@ import 'full_page_embedding_strategy.dart';
 ///   element, provided by the web app programmer through the engine
 ///   initialization.
 abstract class EmbeddingStrategy {
-  EmbeddingStrategy() {
-    // Initialize code to handle hot-restart (debug only).
-    assert(() {
-      _hotRestartCache = HotRestartCacheHandler();
-      return true;
-    }());
-  }
+  EmbeddingStrategy();
```

Comment:
Do we need an empty constructor here?

Reply:
Removing it causes the following error:
```
The class 'FullPageEmbeddingStrategy' can't extend 'EmbeddingStrategy' because 'EmbeddingStrategy' only has factory constructors (no generative constructors), and 'FullPageEmbeddingStrategy' has at least one generative constructor.
```
But I think we can make the subclasses `implements EmbeddingStrategy` instead of extending. Let me try that.

Reply:
Done